crates/storage/libmdbx-rs/src/lib.rs

```diff
@@ -5,7 +5,7 @@
     issue_tracker_base_url = "https://github.com/paradigmxyz/reth/issues/"
 )]
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
-#![allow(missing_docs)]
+#![allow(missing_docs, clippy::needless_pass_by_ref_mut)]
```

Comment:
addressing this lint would make almost all cursor methods non mut, not sure how i feel about that yet

Reply:
why is it bad?

Reply:
because we are actually mutating the cursor (changing its position)